earth_enterprise/src/installer/common.sh

```diff
@@ -0,0 +1,198 @@
+            software_check_retval=1
+        fi
+    else 
+        echo -e "\nThe installer could not determine your machine's operating system."
+        echo -e "Supported Operating Systems: ${SUPPORTED_OS_LIST[*]}\n"
+        software_check_retval=1
+    fi
+
+    return $software_check_retval
+}
+
+determine_os()
+{
+    local retval=0
+    local test_os=""
+    local test_versionid=""
+
+    if [ -f "$OS_RELEASE1" ] || [ -f "$OS_RELEASE2" ]; then
+        if [ -f "$OS_RELEASE1" ]; then
+            test_os="$(cat $OS_RELEASE1 | sed -e 's:\"::g' | grep ^NAME= | sed 's:name=::gI')"
+            test_versionid="$(cat $OS_RELEASE1 | sed -e 's:\"::g' | grep ^VERSION_ID= | sed 's:version_id=::gI')"
+        else
+            test_os="$(cat $OS_RELEASE2 | sed 's:[0-9\.] *::g')"
+            test_versionid="$(cat $OS_RELEASE2 | sed 's:[^0-9\.]*::g')"
+        fi
+
+        MACHINE_OS_FRIENDLY="$test_os $test_versionid"
+        MACHINE_OS_VERSION=$test_versionid
+
+        if [[ "${test_os,,}" == "ubuntu"* ]]; then
+            MACHINE_OS=$UBUNTUKEY
+        elif [[ "${test_os,,}" == "red hat"* ]]; then
+            MACHINE_OS=$REDHATKEY
+        else
+            MACHINE_OS=""
+            echo -e "\nThe installer could not determine your machine's operating system."
+            echo -e "Supported Operating Systems: ${SUPPORTED_OS_LIST[*]}\n"
+            retval=1
+        fi
+    else
+        echo -e "\nThe installer could not determine your machine's operating system."
+        echo -e "Missing file: $OS_RELEASE1 or $OS_RELEASE2\n"
+        retval=1
+    fi
+
+    return $retval
+}
+
+run_as_user() {
+    use_su="su $1 -c 'echo -n 1' 2> /dev/null  || echo -n 0"
+    if [ "$use_su" -eq 1 ] ; then
+        ( cd / ;su $1 -c "$2" )
+    else
+        ( cd / ;sudo -u $1 $2 )
+    fi
+}
+
+show_no_tmp_dir_message()
+{
+    echo -e "\nThe temp install directory specified [$1] does not exist."
+    echo -e "Please specify the path of the extracted install files or first run"
+    echo -e "scons release=1 installdir=$1 install\n"
+}
+
+check_bad_hostname() {
+    if [ -z "$HOSTNAME" ] || [[ " ${BADHOSTNAMELIST[*]} " == *"${HOSTNAME,,} "* ]]; then
+        show_badhostname
+
+        if [ $BADHOSTNAMEOVERRIDE == true ]; then
+            echo -e "Continuing the installation process...\n"
+            return 1
+        else
+            echo -e "Exiting the installer.  If you wish to continue, re-run this command with the -hnf 'Hostname Override' flag.\n"
+            return 0
+        fi
+    fi
+}
+
+show_badhostname()
+{
+    echo -e "\nYour server [$HOSTNAME] contains an invalid hostname value which typically"
+    echo -e "indicates an automatically generated hostname that might change over time."
+    echo -e "A subsequent hostname change would cause configuration issues for the "
+    echo -e "$SOFTWARE_NAME software.  Invalid values: ${BADHOSTNAMELIST[*]}."
+}
+
+check_mismatched_hostname() {
+    if [ $HOSTNAME != $HOSTNAME_F ]; then
```

Comment:
I was testing this and I don't think this function works correctly. The only difference between $HOSTNAME and $HOSTNAME_F is that in $HOSTNAME it converts capital letters to lowercase letters. So, this will return false if there's an uppercase letter in the hostname and true otherwise. It says that it's supposed to test whether the hostname is set to the same as the FQDN but I don't see how it could be doing that as both variables use the hostname -f command. Any thoughts on this? @arjntc1 @asmitaw @dattapm 

```
HOSTNAME="$(hostname -f | tr [A-Z] [a-z] | $NEWLINECLEANER)"
HOSTNAME_F="$(hostname -f | $NEWLINECLEANER)"
```